### PR TITLE
Homework 20 Pull Request

### PR DESCRIPTION
**Pull request by iibrahimsv**
Homework Checking Team,

Homework 20 is ready for review. Please advise 👍🏽

### CODE REVIEW

File: src/test/java/BaseTest.java [hunk 1]
```diff
@@ -1,13 +1,132 @@
...
+    }
+    protected static void addingNewSong() throws InterruptedException {
+
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//tr[@class='song-item']//td[contains(text(),'Epic Song')]"))).click();
+    }
+    protected static void allSongsTab() throws InterruptedException {
+
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//a[@class='songs']"))).click();
```
Comment (Berichicko) [inline src/test/java/BaseTest.java:101]: You can make one method and then use it so as not to repeat yourself, for example

  ```
protected static WebElement waitForElementToBeClickable(WebElement element) {
    return new WebDriverWait(driver, Duration.ofSeconds(10))
        .until(ExpectedConditions.elementToBeClickable(element));
  }
```

image

File: src/test/java/BaseTest.java [hunk 2]
```diff
@@ -1,13 +1,132 @@
...
+    protected static void playlistDeleteBtn() throws InterruptedException {
+
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//button[@title='Delete this playlist']"))).click();
+    }
+    protected static void playlistToDelete() throws InterruptedException {
+
+       WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+       wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//a[@href='#!/playlist/26571']"))).click();
+    }
```
Comment (Berichicko) [inline src/test/java/BaseTest.java:46]: Here is an example using the method `waitForElementToBeClickable(WebElement element)`

 ```
protected static void playlistDeleteBtn() {
        WebElement buttonDelete = driver.findElement(By.xpath("//button[@title='Delete this playlist'"));
        waitForElementToBeClickable(buttonDelete).click();
    }
```
image

**Comment by Berichicko** [comment]
Hello

Homework  20 is cheeked! 

I left some recommendations for the code. 🔥

Please refactoring your code.

Well, great job! 🔥


Your grade: 90  ✊


💖Best wishes, the team HW (Alex)!💖

**Comment by iibrahimsv** [comment]
@Berichicko ,

Thank you for the feedback! I will make the correct changes 👍🏽

**Pull request closed without merging**